provider/kubernetes/client.go

```diff
@@ -200,32 +197,23 @@ func (c *clientImpl) UpdateIngressStatus(namespace, name, ip, hostname string) e
 
 // GetService returns the named service from the given namespace.
 func (c *clientImpl) GetService(namespace, name string) (*corev1.Service, bool, error) {
-	var service *corev1.Service
-	item, exists, err := c.factories[c.lookupNamespace(namespace)].Core().V1().Services().Informer().GetStore().GetByKey(namespace + "/" + name)
-	if item != nil {
-		service = item.(*corev1.Service)
-	}
-	return service, exists, err
+	service, err := c.factories[c.lookupNamespace(namespace)].Core().V1().Services().Lister().Services(namespace).Get(name)
+	exist, err := translateNotFoundError(err)
+	return service, exist, err
```

Comment:
What do you think about just doing

```go
return service, translateNotFoundError(err)
```

here and elsewhere too?

Reply:
@timoreimann I'm afraid my IDE is not allowing me to do so. Is this syntax supported by golang?

Reply:
it's not supported

Reply:
true, my bad.